fix: add retry logic to eval tests (#906)
kyma-project/kyma-companion: tests/blackbox/evaluation.md

    export CONFIG_PATH=<path_to_config-evaluation.json>
    ```

4. (Optional) Configure retry behavior for failed scenarios:

    ```bash
    export KC_EVAL_RETRIES=3  # Default: 3 (total number of attempts including initial attempt)
    ```

>     This controls how many times a failed scenario will be retried. Each retry creates a fresh conversation to handle LLM non-determinism. Set to 1 to disable retries.

Reviewer: ```suggestion
    This configuration controls how many times a failed scenario is retried. Each retry initiates a new conversation to handle LLM non-determinism. Set to 1 to disable retries.
```
